enhancements/sig-architecture/132-work-update-only-when-spec-change/README.md

```diff
@@ -0,0 +1,183 @@
+- [issue 690](https://github.com/open-cluster-management-io/ocm/issues/690) is the case that user wants
+to create a deployment using manifestwork, but what HPA on the spoke the control the replicas of the deployment.
+Hence the user wants the work-agent to ignore the replicas field of the deployment in the manfiestwor if it is
+set.
+
+
+## Proposal
+
+We would like to introduce new options in `Update` and `ServerSideApply` update strategy to resolve the above issue.
+So user can set the option in the manifestwork, so when another actor on the spoke cluster update the resource. The
+work-agent will ignore the change and not try to change the resource back. On the other hand, when the resource spec
+defined in the manifestwork is updated by the user, the work-agent will still update the resource accordingly. In
+summary, the option is to ignore the change triggered from spoke cluster only.
+
+In addition, in `ServerSideApplly` strategy, we would also introduce a new `ignoreDifferences` options similar as what
+is defined in argoCD (https://argo-cd.readthedocs.io/en/stable/user-guide/diffing/). Such that user can choose to not
+let work-agent patch certain resource fields.
+
+### Design Details
+
+#### API change
+
+The change will be added into the `updateStratey` field. For `Update` strategy, we will add an option
+
+```go
+type UpdateConfig struct {
+    // OnSpokeChange defines whether the resource should be overriden by the manifestwork it is changed
+    // on the spoke by another actor.
+    // +kubebuilder:default=Override
+    // +kubebuilder:validation:Enum=Override;NoOverride
+    // +kubebuilder:validation:Required
+    // +required
+	OnSpokeChange string `json:"onSpokeChange,omitempty"`
+}
+```
+
+For existing `ServerSideApply` strategy, we will add:
+
+```go
+type ServerSideApplyConfig struct {
+	...
+	// IgnoreFields defines a list of json paths in the resource that will not be updated on the spoke.
+	IgnoreFields []string `json:"ignoreFields,omitempty"`
+
+	// OnSpokeChange defines whether the resource should be overriden by the manifestwork it is changed
+	// on the spoke by another actor.
+	// +kubebuilder:default=Override
+	// +kubebuilder:validation:Enum=Override;NoOverride
+	// +kubebuilder:validation:Required
+	// +required
+	OnSpokeChange string `json:"onSpokeChange,omitempty"`
+}
+```
+
+#### agent implemntation
+
+When work-agent identity that the `OnSpokeChange` is set `NoOverride` for a certain resource in the `ManifestWork`, the agent
+when apply the resource to the spoke cluster will add an annotation with the key `open-cluster-management.io/object-hash`.
+The value of the annotation is the computed hash of the resource spec in the `ManifestWork`. Later when another actor
+updates the resource, the work-agent will at first check if the object-hash mismatches with the current resource spec
+in the `ManifestWork`. If it is the same, the resource will not be updated so the change from spoke is ignored. When
```

Comment:
Should we explain the behavior of "Delete" when `OnSpokeChange` is set to `NoOverride` here? It seems when the resource is deleted, object-hash is removed and the resource will be applied again?

Reply:
the resource itself will be deleted, so does the annotation. The annotation will be kept if the deleteOption is Orphan, and in this case, I do not think we need to remove the annotation.